lib/models/Cat.js

```diff
@@ -0,0 +1,80 @@
+const pool = require('../utils/pool');
+
+module.exports = class Cat {
+  id;
+  breed;
+  playfulness;
+  intelligence;
+  origin;
+
+  constructor(row) {
+    this.id = row.id;
+    this.breed = row.breed;
+    this.playfulness = row.playfulness;
+    this.intelligence = row.intelligence;
+    this.origin = row.origin;
+  }
+
+  static async getAll() {
+    const { rows } = await pool.query('SELECT * from cats');
+    return rows.map((cat) => new Cat(cat));
+  }
+
+  static async getById(id) {
+    const { rows } = await pool.query(
+      `
+    SELECT * from cats 
+    WHERE 
+    id = $1
+    `,
+      [id]
+    );
+    return new Cat(rows[0]);
```

Comment:
This is where you'd add a check to ensure you're building a Cat instance with valid data.

```js
if (!rows[0]) return null;
return new Cat(rows[0]);